app/Http/Middleware/AdminMiddleware.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+
+class AdminMiddleware
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if(auth()->user()->admin()) {
```

Comment:
The methods `admin()`, `customer()` and `dataClerk()` are not straightforward. `auth()->user()->customer()` could also mean a relationship or scope. I would suggest something more clear like `isAdmin()`, `isCustomer()` and `isDataClerk` since we are dealing with nouns. For verbs such as `invoiced()`, `sold()` etc do not need to prefix with verb. So here, `sold()` is already more clear of the outcome.